### ISSUE TITLE
the  loss function formula

### ISSUE THREAD

**Issue by neda-noor-m** [cvlab-tohoku/DSEBImageInpainting#2]
Hello 
I've tried to re-implement your paper entitled "Deep Structured Energy-Based Image Inpainting" for inpainting problem. However, I could not understand the loss function which you have mentioned in the 10-th line of the Algorithm 1 (which is used for updating $$\theta$$). Please explain it in more details. Specifically, I do not understand how you bring the constraint into the loss function. I think $$y_hat$$ does not depend on theta. Since it is the result of the previous gradient descent on the energy function (line 7). So, it does not make sense to perform optimization on the loss function which is only depend on constants. Please clearly written the loss function with all terms which might be used for it.

**Comment by fazilaltinel** [comment]
Hello,

$ \hat{y} $ is not a constant. Therefore, the loss function $ \ell(\hat{y}, y) $ can be minimized with respect to $ \Theta $. We explain how loss minimization is performed in Section 3.B in our paper (Equations 3 and 5). Please check it for more detail.

**Comment by neda-noor-m** [comment]
Dear Mr. Altinel,

I read your paper several times, but unfortunately I couldn't completely understand some details of your algorithm 1. The main problem is that do you use “cost-augmented inference” to update y (line 7 )? Why is the value of energy function (E_x) updated in line 8 of Algorithm 1? The The updated energy in which of the formula is used? Is your algorithm exactly similar to the original spen paper (belanger 2016)?  please explain these in more detail.
In my implementation, I updated y using “cost-augmented inference”. But unfortunately, I got not reasonable results. I use the Olivetti face dataset.
